Debugged Interfaces + Issue163 Base Circuits

rubenbaetens opened the pull request:
Current pull requests focuses on two things:
* The adoption of the Fluid.BaseCircuits having basic thermodynamic circuits based on Hysopt
* The debugging of the Interfaces which appeared after running BESTEST

Mathadon commented on IDEAS/Controls/Control_fixme/package.mo:
```diff
@@ -0,0 +1,17 @@
+within IDEAS.Controls;
```
This package was deleted earlier. Not sure why git merges it in again. I'll remove it again.

Mathadon commented on IDEAS/Electric/Distribution/BaseClasses/Branch.mo:
```diff
@@ -24,7 +24,7 @@ model Branch
 equation
   v = Z*i;
-  Plos = R*Modelica.ComplexMath.'abs'(i)^2;
+  Plos = R*Modelica.ComplexMath.'abs'(i)^2+1;
```
Is this correct?

arnoutaertgeerts replied:
No :)

Mathadon commented on IDEAS/Electric/Distribution/BaseClasses/Branch.mo:
```diff
@@ -12,7 +12,7 @@ model Branch
       Placement(transformation(extent={{10,10},{30,30}}), iconTransformation(
           extent={{10,10},{30,30}})));
-  Modelica.Thermal.HeatTransfer.Sources.PrescribedHeatFlow prescribedHeatFlow
+  Modelica.Thermal.HeatTransfer.Sources.PrescribedHeatFlow prescribedHeatFlow(alpha=1)
```
@arnoutaertgeerts the reason why you needed to add this is the following:
when the heatport is unconnected externally this forces Q_flow = 0. However Q_flow = Plos so this results into a singular system. Setting alpha =/= 0 allows to solve this because the port temperature will increase when the power cannot be delivered. This is however not physical.
I propose to make the calculation of Plos conditional: add a boolean parameter to enable the heatport/prescribedheatflow and the calculation of Plos. When enabling this parameter port_a must be connected.

rubenbaetens replied:
@arnoutaertgeerts @Mathadon if we work conditionally, note that you only make the thermal connection conditional, and **not** the calculation of Plos, cause knowing the resistive losses are key in models.

Mathadon commented on IDEAS/Electric/Distribution/BaseClasses/BranchLenTyp.mo:
```diff
@@ -2,7 +2,8 @@ within IDEAS.Electric.Distribution.BaseClasses;
   extends IDEAS.Electric.Distribution.BaseClasses.Branch(R=typ.RCha*mulFac*len,
-      X=typ.XCha*mulFac*len);
+      X=typ.XCha*mulFac*len,
+    prescribedHeatFlow(alpha=1));
```
See note above

arnoutaertgeerts commented:
@rubenbaetens can I delete the following models?
- CollectorUnit
- Overflow
- PipeSplit_Alternative

I'm not sure what they are for :)

rubenbaetens commented:
@arnoutaertgeerts i changed the overflow and deleted it, for me the others, pipesplit_alternative can be gone but CollectorUnit i should keep, i use it often, don't see why it should be obsolete.

arnoutaertgeerts commented:
@rubenbaetens Why do you need the CollectorUnit model? Can't you just use the fluid connectors to achieve the same behavior?

rubenbaetens commented:
What's left on the ToDo-list here ?

rubenbaetens commented:
Thx for the effert @Mathadon.
